Cxl oauth integration

mmarifat opened the pull request:
## Description

<!-- Describe the proposed solution/fix/feature in this pull request here. -->

This PR solves # .
Login with Wordpress via oauth2
Detailed documentation in in OAUTH-DOC.md

### Steps completed:

<!-- Check all completed steps so we know  -->

According to our definition of done, I have completed the following steps:

- [x] Acceptance criteria met
- [x] Unit tests added
- [x] Docs updated (including config and env variables)
- [x] Translations added
- [x] UX tested
- [x] Browsers / platforms tested
- [x] Rebased & ready to merge without conflicts
- [x] Reviewed own code

mmarifat committed:
feat: jwp-cxl oauth integration

mmarifat committed:
feat: oAuth documentation

mmarifat committed:
chore: doc update for custom parameters

mmarifat committed:
fix: snapshot Cinema test

lkraav reviewed (commented):
No idea, this is way too big to grok at the bandwidth I have available.

pawelkmpt commented on packages/common/types/account.ts:
```diff
@@ -164,6 +164,7 @@ export type Customer = {
   metadata: Record<string, unknown>;
   lastName?: string;
   fullName?: string;
+  isPremium?: boolean;
```
This seems to be risky in terms of getting future updates from an upstream repository.

At the same time I see `metadata` field above, what seems to me like a flexible property to add custom values. What can you tell me about it?

mmarifat replied:
the interface supports any key as typed as [x:string], I have put this to get type suggestion, can be removed too

pawelkmpt commented on packages/hooks-react/src/useOAuth.ts:
```diff
@@ -0,0 +1,88 @@
+    },
+    redirectUri: (env.APP_OAUTH_REDIRECT_URL || undefined) as string,
+    // eslint-disable-next-line no-alert
+    onRefreshTokenExpire: (event: TRefreshTokenExpiredEvent) => window.confirm('Session expired. Refresh page to continue using the site?') && event.login(),
```
```suggestion
    onRefreshTokenExpire: (event: TRefreshTokenExpiredEvent) => window.confirm('Session expired. Refresh page to continue using the site.') && event.login(),
```

mmarifat replied:
Sorry, you can change to any text

pawelkmpt commented on packages/ui-react/src/components/Header/Header.tsx:
```diff
@@ -106,7 +111,8 @@ const Header: React.FC<Props> = ({
   });
 
   // only show the language dropdown when there are other languages to choose from
-  const showLanguageSwitcher = supportedLanguages.length > 1;
+  // FEAT:: no language switcher in oauth mode
+  const showLanguageSwitcher = !isOAuthMode && supportedLanguages.length > 1;
```
Where supported languages can be defined?

mmarifat replied:
in the .env

pawelkmpt commented on packages/ui-react/src/components/Header/Header.tsx:
```diff
@@ -135,49 +141,53 @@ const Header: React.FC<Props> = ({
   const renderUserActions = () => {
     if (!canLogin || breakpoint <= Breakpoint.sm) return null;
-
+    // FEAT:: back to main account cta if oauth mode
```
Did you consider creating custom component `CXLHeader` based on the `Header` instead of heavily modifying the upstream one?

mmarifat replied:
yeah, sounds good, should have thought like others

pawelkmpt reviewed (commented):
Just a few questions

mmarifat committed:
perf: early return
perf: remove isPremium from interface

mmarifat committed:
Merge branch 'cxl-develop' of github.com:conversionxl/ott-web-app into cxl-oauth-integration

# Conflicts:
#	packages/ui-react/src/components/Header/Header.tsx
#	packages/ui-react/src/containers/Layout/Layout.tsx

mmarifat committed:
chore: 16px variable margin left in header cta
chore: 20px margin top in mobile sidebar item
chore: api calls only when token change
chore: user object pushing before me api calls to stop flicking

mmarifat committed:
chore: ini for searchable app

mmarifat committed:
feat: favorite local service
feat: continue watching local service

mmarifat committed:
revert: app_content_search custom parameter remove
